test/functional/api/v2/deployments_controller_test.rb

```diff
@@ -2,17 +2,43 @@
 
 class ForemanDeployments::Api::V2::DeploymentsControllerTest < ActionController::TestCase
   class FakeTask < ForemanDeployments::Tasks::BaseDefinition
+    def validate
+      ForemanDeployments::Validation::ValidationResult.new
+    end
   end
 
-  describe 'creating a deployment' do
-    setup do
-      @stack = ForemanDeployments::Stack.create(
-        :name => 'stack1',
-        :definition => 'SomeTask:!Task'
-      )
-      @deployment_params = { :name => 'A Deployment', :stack_id => @stack.id }
+  class InvalidFakeTask < ForemanDeployments::Tasks::BaseDefinition
+    def validate
+      ForemanDeployments::Validation::ValidationResult.new([
+        'Some validation error'
+      ])
     end
+  end
 
+  setup do
+    @stack = ForemanDeployments::Stack.create(
+      :name => 'stack1',
+      :definition => [
+        'Task1: !task:FakeTask',
+        '  param1: hardcoded',
+        'Task2: !task:FakeTask'
+      ].join("\n")
+    )
+    @deployment_params = { :name => 'A Deployment', :stack_id => @stack.id }
+    @deployment = ForemanDeployments::Deployment.create(
+      :name => 'Another deployment',
+      :configuration => ForemanDeployments::Configuration.new(:stack => @stack)
+    )
+
+    ForemanDeployments.registry.register_task(FakeTask)
```

Comment:
Same as in #13, please avoid changing global setting in a single test. It's better to isolate the setting.